driver/driver.go

```diff
@@ -214,9 +216,14 @@ func NewReaderWithConfig(ctx context.Context, conf Config, program ast.Proc, zct
 		cancel()
 		return nil, err
 	}
-	return &muxReader{
+	mr := &muxReader{
 		cancel:      cancel,
 		muxOutput:   mux,
 		statsTickCh: make(chan time.Time),
-	}, nil
+		zr:          ioutil.NopCloser(nil),
+	}
+	if _, ok := reader.(io.Closer); ok {
```

Comment:
I'm not seeing why the changes required added this type assertion, can you briefly explain?

Reply:
The reader might be a `multipart.Part` which needs to be closed.
One alternate implementation would be to change the signature to pass a `zbuf.ReadCloser` instead, and do away with the type assertion.